### PR TITLE
disallow non-Superusers to manage admins

### PR DESCRIPTION
**Pull request by michield**
## Description
non Superusers should not be allowed to manage admins, even when they have "config" permissions

### CODE REVIEW

**Comment by michield** [comment]
Hmm, how annoying the CI fails again.

File: public_html/lists/admin/connect.php [hunk 1]
```diff
@@ -808,6 +801,17 @@ function pageTitle($page)
     $GLOBALS['pagecategories']['system']['pages'][] = 'update';
     $GLOBALS['pagecategories']['system']['menulinks'][] = 'update';
 }
+if (isSuperUser()) {
+  foreach (array('admins','admin','importadmin','adminattributes') as $adminPage) {
+    $GLOBALS['pagecategories']['config']['menulinks'][] = $adminPage;
```
Comment (bramley) [inline public_html/lists/admin/connect.php:806]: This is adding an extra page to the Config menu. Currently that menu does not include the admin page.

File: public_html/lists/admin/connect.php [hunk 2]
```diff
@@ -808,6 +801,17 @@ function pageTitle($page)
     $GLOBALS['pagecategories']['system']['pages'][] = 'update';
     $GLOBALS['pagecategories']['system']['menulinks'][] = 'update';
 }
+if (isSuperUser()) {
```
Comment (bramley) [inline public_html/lists/admin/connect.php:804]: When you login the displayed Config menu does not include these items. They are shown only when you display a page after logging-in. I noticed that for other menu items in the past but never tried to figure out why.

The reason is that these statements are run before the login action has been processed, so the admin isn't a super user at that point.

One way to fix this is to redirect on successfully logging-in. I can do a pull request for that.

File: public_html/lists/admin/connect.php [hunk 3]
```diff
@@ -1103,7 +1107,7 @@ function topMenu()
         ) {
             continue;
         }
-
+var_dump($category);exit;
```
Comment (bramley) [inline public_html/lists/admin/connect.php:1110]: Remove this.

Comment (michield) [reply public_html/lists/admin/connect.php:1110]: Ah, yes, that's oversight.

File: public_html/lists/admin/connect.php (hunk 1 above)
Comment (michield) [reply public_html/lists/admin/connect.php:806]: Yes, but that's not a big issue.

File: public_html/lists/admin/connect.php (hunk 3 above)
Comment (michield) [reply public_html/lists/admin/connect.php]: removed

**Comment by michield** [comment]
> This looks ok. It is removing the ability for an ordinary admin to edit/view their own admin details though. I don't use ordinary admins so don't know how important that might be. Perhaps that can be put back later but removing the current ability to change their own privileges.

Yes, for now this resolves a security issue where a sub-admin can update the details of a superuser, which is bad. We can add that later again.

File: public_html/lists/admin/connect.php (hunk 2 above)
Comment (michield) [reply public_html/lists/admin/connect.php:804]: Interesting, I haven't noticed that.

Comment (michield) [reply public_html/lists/admin/connect.php:804]: Ah, interesting, that must be the reason that the CI jobs failed. First time you login, it's not there. I've commented them out for now, but once you have a PR to fix that (or I) we can add that again

Comment (bramley) [reply public_html/lists/admin/connect.php:804]: My idea of redirecting on successful login works for the browser but not for the REST API plugin. 
Firstly, the client needs to have enabled CURLOPT_FOLLOWLOCATION which currently isn't needed, so there is a possibility of breaking a client. The sample client doesn't do that, https://github.com/michield/phplist-restapi-client

Returning a 302 code will make the client request the page again using GET, but the REST API plugin treats a GET as a documentation page request.
Returning a 307 code will make the client request the page using the original method of POST. This does work when CURLOPT_FOLLOWLOCATION is enabled.

Comment (michield) [reply public_html/lists/admin/connect.php:804]: The reason it takes two page loads is that accesscheck.php defines the "isSuperuser" and it only gets included in home.php after the header (with the menu) is generated. In a way, I'm puzzled it works at all, as isSuperuser should really be at a higher level. And then it does get set on first login in https://github.com/phpList/phplist3/blob/main/public_html/lists/admin/index.php#L319

Ah, but then "connect.php" has already been included, so the menu additions won't get added at that stage. So, I'll move that down a bit.

**Comment by michield** [comment]
This commit will make the items appear in the menu on first load: https://github.com/phpList/phplist3/pull/986/commits/774a075af6761355b104026aad65a55abcac8a7e

**Comment by bramley** [comment]
Yes, that has fixed the problem. Now the four menu items appear after logging-in.

**Review by aulona1 — APPROVED** [review]

**Pull request merged**